### PR TITLE
Fetch most wanted items from trakt

### PR DESCRIPTION
**Pull request by davidemarcoli**
# Pull Request Check List

Resolves: #issue-number-here

- [ ] Added **tests** for changed code.
- [ ] Updated **documentation** for changed code.

### CODE REVIEW

File: src/program/content/trakt.py [hunk 1]
```diff
@@ -73,18 +76,33 @@ def missing(self):
             logger.log("TRAKT", "Trending fetching is disabled.")
         if not self.settings.fetch_popular:
             logger.log("TRAKT", "Popular fetching is disabled.")
+        if not self.settings.fetch_most_watched:
+            logger.log("TRAKT", "Most watched fetching is disabled.")
 
     def run(self):
         """Fetch media from Trakt and yield Movie, Show, or MediaItem instances."""
+        global last_update
```
Comment (dreulavelle) [inline src/program/content/trakt.py:84]: I was thinking a class attr, like `self.last_update` lol. I always though `global` was gross 😅

File: src/program/content/trakt.py [hunk 2]
```diff
@@ -73,18 +76,33 @@ def missing(self):
...
 
         watchlist_ids = self._get_watchlist(self.settings.watchlist) if self.settings.watchlist else []
         collection_ids = self._get_collection(self.settings.collection) if self.settings.collection else []
         user_list_ids = self._get_list(self.settings.user_lists) if self.settings.user_lists else []
-        trending_ids = self._get_trending_items() if self.settings.fetch_trending else []
-        popular_ids = self._get_popular_items() if self.settings.fetch_popular else []
+
+        # Check if it's the first run or if a day has passed since the last update
+        current_time = datetime.now()
+        if last_update is None or (current_time - last_update) > timedelta(days=1):
```
Comment (dreulavelle) [inline src/program/content/trakt.py:92]: I was thinking to just add this to those functions instead, but this works too 😁

File: src/program/content/trakt.py (hunk 1 above)
Comment (davidemarcoli) [reply src/program/content/trakt.py:84]: 👍🏻 Could you change it? Otherwise I'll do it in a few hours.

File: src/program/settings/models.py [hunk 3]
```diff
@@ -174,6 +174,9 @@ class TraktModel(Updatable):
     trending_count: int = 10
     fetch_popular: bool = False
     popular_count: int = 10
+    fetch_most_watched: bool = False
+    most_watched_period: str = "weekly"
```
Comment (dreulavelle) [inline src/program/settings/models.py:178]: do we want to set daily by default?

File: src/program/content/trakt.py [hunk 4]
```diff
@@ -73,18 +77,31 @@ def missing(self):
             logger.log("TRAKT", "Trending fetching is disabled.")
         if not self.settings.fetch_popular:
             logger.log("TRAKT", "Popular fetching is disabled.")
+        if not self.settings.fetch_most_watched:
+            logger.log("TRAKT", "Most watched fetching is disabled.")
```
Comment (dreulavelle) [inline src/program/content/trakt.py:81]: Honestly, we can ditch these 3 log messages, as they dont really provide any value to the user. I used to use them before as a way to show users whats available.. we can remove these 3 log messages though 👌

File: src/program/content/trakt.py [hunk 5]
```diff
@@ -13,6 +14,8 @@
 from utils.ratelimiter import RateLimiter
 from utils.request import get, post
 
+# Global variable to track the last update time
+last_update = None
```
[the remark this answers is not in the document]
Comment (davidemarcoli) [reply src/program/content/trakt.py:18]: Ah, missed it.. Thanks!

**Review by dreulavelle — APPROVED** [review]

**Pull request merged**